src/analytics/trackingEvents.ts

```diff
@@ -0,0 +1,25 @@
+// list of tracking events
+export enum generics {
+  pressedButton = 'Pressed Button',
+  pressedButton2 = 'Pressed Button 2',
+}
+
+export enum swaps {
+  submittedSwap = 'Submitted Swap',
+}
+
+// how im thinking of improving readability and introducing automated category tagging
+export const TrackingEvents = { ...generics, ...swaps };
```

Comment:
Do we need categories? Why not do these as a single `enum`?

We could also consider how we'll consume these as well. If we re-export these from the `index.ts` file, we can get easy namespacing i.e. `export * as TrackingEvents from '@/analytics/trackingEvents'` becomes:

```
import { TrackingEvents } from '@/analytics'

TrackingEvents.swaps.submittedSwap
```

But honestly I think a single `enum` would be fine here.